tfjs-converter/src/executor/graph_executor.ts

```diff
@@ -252,10 +272,14 @@ export class GraphExecutor implements FunctionExecutor {
               outputNodeNames, intermediateTensorConsumerCount);
         }
       }
+      if (this.keepIntermediateTensors) {
+        this.clonedTensorsMap = this.cloneTensorMap(tensorsMap);
+      }
```

Comment:
This is possible. But 1), this.cloneTensorMap can not be reused, we need some new logic to do this.
2), Your above mentioned tensorMapForEach refactor is also impacted.

Reply:
I think Matt is right. For the original intermediate tensors, we should clone each of them when we get them in L274. Otherwise, we are leaking them due to the changes (L330) you add in `this.checkTensorForDisposal`. Since you already use cloned tensor, theoretically, `this.checkTensorForDisposal` should be unchanged.  And the code logic you mentioned can be changed based on our requirement.
Similar for the async execute.